_api/hasura/cron/syncCoSouls.ts

```diff
@@ -148,76 +153,96 @@ const getCoSoulsToUpdate = async () => {
 };
 
 const syncCoSoulToken = async (
-  coSoulId: number,
-  address: string,
-  totalPGIVE: number,
-  tokenId: number
+  updatedCosouls: { cosoul: CoSoul; localPGIVE: number }[]
 ) => {
-  if (totalPGIVE > 0) {
-    totalPGIVE = Math.floor(totalPGIVE);
-    await setOnChainPGIVE(tokenId, totalPGIVE);
-    console.log(
-      'set PGIVE on chain for tokenId: ' +
-        tokenId +
-        ' address: ' +
-        address +
-        ' to ' +
-        totalPGIVE
-    );
-  } else {
-    console.log(
-      'skipping setting on-chain PGIVE because it is 0 for tokenId: ' +
-        tokenId +
-        ' address: ' +
-        address
-    );
+  let payload = paddedHex(PGIVE_SLOT, 2, true); //1byte for slot
```

Comment:
can we make all paddedHex stuff hidden away inside a helper function in the cosoul api? @sshmm

Reply:
@crabsinger done getPayload will take tokenId and pgives and returns 16 characters hex for them